L1Trigger/L1TGlobal/plugins/TriggerMenuParser.cc

```diff
@@ -172,7 +182,30 @@ void l1t::TriggerMenuParser::setGtAlgorithmAliasMap(const AlgorithmMap& algoMap)
 
 std::map<std::string, unsigned int> l1t::TriggerMenuParser::getExternalSignals(const L1TUtmTriggerMenu* utmMenu) {
   using namespace tmeventsetup;
-  const esTriggerMenu* menu = reinterpret_cast<const esTriggerMenu*>(utmMenu);
+  const std::map<std::string, L1TUtmCondition>& condMap = utmMenu->getConditionMap();
+
+  std::map<std::string, unsigned int> extBitMap;
+
+  //loop over the algorithms
+  for (std::map<std::string, L1TUtmCondition>::const_iterator cit = condMap.begin(); cit != condMap.end(); cit++) {
+    const L1TUtmCondition& condition = cit->second;
+    if (condition.getType() == esConditionType::Externals) {
+      // Get object for External conditions
+      const std::vector<L1TUtmObject>& objects = condition.getObjects();
+      for (size_t jj = 0; jj < objects.size(); jj++) {
+        const L1TUtmObject object = objects.at(jj);
+        if (object.getType() == esObjectType::EXT) {
+          unsigned int channelID = object.getExternalChannelId();
+          std::string name = object.getExternalSignalName();
+
+          if (extBitMap.count(name) == 0)
+            extBitMap.insert(std::map<std::string, unsigned int>::value_type(name, channelID));
+        }
+      }
+    }
+  }
+
+  /*   const esTriggerMenu* menu = reinterpret_cast<const esTriggerMenu*>(utmMenu);
```

Comment:
please delete commented code, or give some explanation why it's still there

Reply:
This has been removed in the next update